Docs: Add Custom Code Guide

dericksozo opened the pull request:
(docs): Adding a new guide around adding custom code / business logic to your service after building it. It clarifies the process of merging branches, modifying files, and more.

PazYanoverr commented on docs/how-to/add-custom-code.md:
```diff
@@ -1,95 +1,78 @@
+# Understanding Custom Code in Amplication
 
-Although your application built with Amplication is fully functional and you can start using it as it was shipped, you probably want to add your core business logic and other functionality to your server.
+This page provides an overview of how custom code works in Amplication and our vision for seamless integration between generated and custom code.
```
I think it's not a vision but reality...

PazYanoverr commented on docs/how-to/add-custom-code.md:
```diff
@@ -1,95 +1,78 @@
-### The 'base' folder
+Amplication may still update the non-base files that include your custom code in certain circumstances. These updates are limited to changes necessary for preventing errors and ensuring the project can build correctly. For example, if you remove a plugin that was previously used in your service, Amplication might update the non-base files to remove references to the removed plugin, thus preventin ...
+
+This approach allows Amplication to maintain the integrity of your project structure while still preserving your custom code to the greatest extent possible.
```
I think we need to emphasize here that Amplication code will always respect the user's custom code, and won't overwrite it.

PazYanoverr commented on docs/how-to/add-custom-code.md:
```diff
@@ -1,95 +1,78 @@
+4. **Conflict Resolution**: If conflicts arise, Amplication provides clear indications and allows you to resolve them manually.
 
-The base folder contains all the base files that are automatically generated by Amplication with every change. These files should not be altered or customized as they will be overwritten and the changes will not be preserved.
+This approach allows you to freely add custom business logic, new endpoints, or any other customizations while still benefiting from Amplication's code generation and updates.
```
```suggestion
This approach allows you to freely add custom business logic, or any other customizations while still benefiting from Amplication's code generation and updates.
```

PazYanoverr commented on docs/how-to/add-custom-code.md:
```diff
@@ -1,95 +1,78 @@
+## Considerations
 
-## Examples
+- Customizing the client application (Admin UI) code is not currently supported within Amplication's regeneration process. If you need to customize the client, it's recommended to clone the entire **Admin** folder to a separate repository.
```
Not so sure about it. Are you sure?

PazYanoverr commented on docs/getting-started/add-custom-code.md:
```diff
@@ -0,0 +1,246 @@
+2. [Set up entities](/set-up-entities/) for your service
+3. [Configured roles and permissions](/configure-roles-and-permissions/)
+4. [Added plugins to your service](/add-plugins-service/)
+5. [Committed changes and built a new version](/commit-and-build-new-versions/)
```
Maybe another "prereq." is the "Understanding custom code in Amplication" doc?

dericksozo commented on docs/getting-started/add-custom-code.md:
```diff
@@ -0,0 +1,246 @@
+
+## Step 7: Create a Pull Request
+
+Go to your repository on GitHub (or your chosen git provider) and create a new pull request:
```
Regarding the importance of detailing those git-focused steps, I can simplify it a bit, but I'm also addressing the question this developer raised in [this issue](https://github.com/orgs/amplication/projects/5/views/1?pane=issue&itemId=78576046).

PazYanoverr commented on docs/how-to/add-custom-code.md:
```diff
@@ -48,31 +48,39 @@ src
+## Handling Conflicts
 
-4. **Conflict Resolution**: If conflicts arise, Amplication provides clear indications and allows you to resolve them manually.
+While Amplication strives to preserve your custom code, conflicts may arise, especially with significant changes to your data model or entity structure. If conflicts occur:
```
Let's emphasize that the conflict resolution should take place on 'amplication' branch, and then- merge the result to the main branch

PazYanoverr commented on docs/how-to/add-custom-code.md:
```diff
@@ -48,31 +48,39 @@ src
-- While Amplication strives to maintain compatibility, major changes to your data model or entity structure may require manual updates to your custom code.
+- While all code can be customized, we recommend focusing custom code in the non-base files for easier maintenance.
+- Major changes to your data model or entity structure may require manual updates to your custom code.
+- Client-side customization (Admin UI) is supported, but changes may not be automatically merged in future builds. Consider maintaining a separate repository for extensive client-side customizations.
```
Let's remove it. I don't think it adds value here

PazYanoverr commented on docs/how-to/add-custom-code.md:
```diff
@@ -1,95 +1,78 @@
 ---
 
-# Add custom code to your services
+# Understanding Custom Code in Amplication
```
Adding to this comment-
Not sure I understand why we have two separate pages for the custom code. It seems like there is redundancy between the files and the content repeats itself.
I can understand the need to have one page to explain the concepts of the generated code, and custom code, and the other to show an example of updating custom code or such but then- we need to make sure each page has its own "uniqueness" and contribute different things to the subject.
WDYT?

dericksozo commented:
Hey @PazYanoverr, I implemented your latest feedback. Also, the additional feedback you gave me on the call really helped and I've implemented that too. The pages have their own uniqueness, with one focusing on implementation and the other on explanation, as suggested.

My one recommendation is adding a Conclusion section to the end of the implementation guide. Now that the user has this custom code full name retrieval method, what can they do with that? Can they use it in the app, or somewhere else in their code? What are the repercussions? Can we link them to a GitHub repo for a more sophisticated example?

yuval-hazaz commented:
@dericksozo this one looks good to me - please resolve the conflict and I will approve and merge it
